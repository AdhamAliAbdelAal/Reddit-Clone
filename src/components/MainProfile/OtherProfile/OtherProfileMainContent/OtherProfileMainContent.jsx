import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import UserProvider from '../../../../contexts/UserProvider';
import OtherProfileContent from './OtherProfileContent/OtherProfileContent';
import OtherProfilePostsTap from './OtherProfilePostsTap/OtherProfilePostsTap';
import OtherProfileSidebar from './OtherProfileSidebar/OtherProfileSidebar';
import { ProfilePage } from './styles';

const renderSwitch = (param) => {
  if (param === undefined || param === 'sort=new' || param === 'sort=hot' || param === 'sort=top' || param === 'sort=top&t=day') {
    return (
      <OtherProfileContent />
    );
  }

  if (param === 'submitted') {
    return (
      <OtherProfilePostsTap />
    );
  }
  return (
    <div />
  );
};

/**
 * OtherProfileMainContent if the profile page
 *
 * @component MainContentProfile
 * @property  {function} handleClick to navigate to the subpage
 * @return {React.Component} - OtherProfileMainContent
 */

function OtherProfileMainContent() {
  const { subTitle, username } = useParams();

  return (
    <Box>
      <UserProvider name={username}>
        <ProfilePage>
          {renderSwitch(subTitle)}
          <OtherProfileSidebar />
        </ProfilePage>
      </UserProvider>
    </Box>
  );
}

export default OtherProfileMainContent;
