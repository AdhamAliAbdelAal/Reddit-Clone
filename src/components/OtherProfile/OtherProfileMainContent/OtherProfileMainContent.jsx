import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import ContentProvider from '../../../contexts/ContentProvider';
import PostsProvider from '../../../contexts/PostsProvider';
import UserProvider from '../../../contexts/UserProvider';
import UserInfoProvider from '../../../contexts/UserInfoProvider';
import OtherProfileContent from './OtherProfileContent/OtherProfileContent';
import OtherProfilePostsTap from './OtherProfilePostsTap/OtherProfilePostsTap';
import OtherProfileSidebar from './OtherProfileSidebar/OtherProfileSidebar';
import { ProfilePage } from './styles';
import CommunitiesProvider from '../../../contexts/CommunitiesProvider';

const renderSwitch = (param, username) => {
  if (param === undefined || param === 'sort=new' || param === 'sort=hot' || param === 'sort=top' || param === 'sort=top&t=day') {
    return (
      <ContentProvider name={username}>
        <OtherProfileContent />
      </ContentProvider>
    );
  }

  if (param === 'submitted') {
    return (
      <PostsProvider name={username}>
        <OtherProfilePostsTap />
      </PostsProvider>
    );
  }
  return (
    <div />
  );
};

/**
 * OtherProfileMainContent if the profile page
 *
 * @component MainContentProfile
 * @property  {function} handleClick to navigate to the subpage
 * @return {React.Component} - OtherProfileMainContent
 */

function OtherProfileMainContent() {
  const { subTitle, username } = useParams();
  return (
    <Box>
      <UserProvider name={username}>
        <ProfilePage>
          {renderSwitch(subTitle, username)}
          <UserInfoProvider>
            <CommunitiesProvider>
              <OtherProfileSidebar />
            </CommunitiesProvider>
          </UserInfoProvider>
        </ProfilePage>
      </UserProvider>
    </Box>
  );
}

export default OtherProfileMainContent;
