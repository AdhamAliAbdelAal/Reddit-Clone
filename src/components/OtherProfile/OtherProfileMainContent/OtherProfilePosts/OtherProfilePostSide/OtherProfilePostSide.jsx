import { useState } from 'react';
import {
  UpArrow, DownArrow, SidebarQueueBox, UpArrowFilled, DownArrowFilled, Number,
} from './styles';

/**
 * Sidebar of the post conatining arrow up and down
 *
 * @component OtherProfilePostSide
 * @property {number} points -number of votes on a post
 * @property {number} postVoteStatus -is the post upvoted or down voted or none
 * @returns {React.Component} OtherProfilePostSide
 */
function OtherProfilePostSide(props) {
  const { points, postVoteStatus } = props;
  const [postPoints, setPostPoints] = useState(points);

  const handleClickUp = () => {
    setPostPoints(postPoints + 1);
  };
  const handleClickDown = () => {
    setPostPoints(postPoints - 1);
  };
  return (
    <SidebarQueueBox>
      {
        (postVoteStatus === '1') ? <UpArrowFilled onClick={() => { handleClickDown(); }} /> : <UpArrow onClick={() => { handleClickUp(); }} />
      }
      <Number variant="caption" sx={{ fontWeight: 700 }}>{postPoints}</Number>
      {
        (postVoteStatus === '-1') ? <DownArrowFilled onClick={() => { handleClickUp(); }} /> : <DownArrow onClick={() => { handleClickDown(); }} />
      }
    </SidebarQueueBox>
  );
}

export default OtherProfilePostSide;
