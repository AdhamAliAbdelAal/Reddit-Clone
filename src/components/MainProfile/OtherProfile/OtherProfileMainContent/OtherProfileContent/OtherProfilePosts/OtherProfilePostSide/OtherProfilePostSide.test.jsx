import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import OtherProfileMainContent from '../../../OtherProfileMainContent';
import OtherProfilePostSide from './OtherProfilePostSide';

test('test snapshot', async () => {
  const tree = renderer.create(<Router><OtherProfileMainContent><OtherProfilePostSide /></OtherProfileMainContent></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
