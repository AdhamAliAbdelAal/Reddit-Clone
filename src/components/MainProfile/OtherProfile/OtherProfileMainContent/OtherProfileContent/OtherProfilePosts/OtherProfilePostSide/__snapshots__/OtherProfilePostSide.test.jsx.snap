// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-1020iqq"
  >
    <div
      className="MuiBox-root css-su0a71"
    >
      <div
        className="MuiBox-root css-3jhl7o"
      >
        <div
          className="MuiBox-root css-92i8si"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-hph3bd-MuiButtonBase-root-MuiButton-root"
            condition="true"
            data-testid="new-btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="NewReleasesOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m23 12-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-4.51 2.11.26 2.79-2.74.62-1.43 2.41L12 18.82l-2.58 1.11-1.43-2.41-2.74-.62.26-2.8L3.66 12l1.85-2.12-.26-2.78 2.74-.61 1.43-2.41L12 5.18l2.58-1.11 1.43 2.41 2.74.62-.26 2.79L20.34 12l-1.85 2.11zM11 15h2v2h-2zm0-8h2v6h-2z"
                />
              </svg>
            </span>
            New
            <span
              className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="MuiBox-root css-bs36a5"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-hph3bd-MuiButtonBase-root-MuiButton-root"
            condition="true"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="NewReleasesIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m23 12-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-10 5h-2v-2h2v2zm0-4h-2V7h2v6z"
                />
              </svg>
            </span>
            New
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-13ckc73-MuiButtonBase-root-MuiButton-root"
            condition="false"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LocalFireDepartmentOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m16 6-.44.55c-.42.52-.98.75-1.54.75C13 7.3 12 6.52 12 5.3V2S4 6 4 13c0 4.42 3.58 8 8 8s8-3.58 8-8c0-2.96-1.61-5.62-4-7zm-4 13c-1.1 0-2-.87-2-1.94 0-.51.2-.99.58-1.36L12 14.3l1.43 1.4c.37.37.57.85.57 1.36 0 1.07-.9 1.94-2 1.94zm3.96-1.5c.04-.36.22-1.89-1.13-3.22L12 11.5l-2.83 2.78C7.81 15.62 8 17.16 8.04 17.5 6.79 16.4 6 14.79 6 13c0-3.16 2.13-5.65 4.03-7.25.23 1.99 1.93 3.55 3.99 3.55.78 0 1.54-.23 2.18-.66C17.34 9.78 18 11.35 18 13c0 1.79-.79 3.4-2.04 4.5z"
                />
              </svg>
            </span>
            Hot
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-13ckc73-MuiButtonBase-root-MuiButton-root"
            condition="false"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloudUploadOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM19 18H6c-2.21 0-4-1.79-4-4 0-2.05 1.53-3.76 3.56-3.97l1.07-.11.5-.95C8.08 7.14 9.94 6 12 6c2.62 0 4.88 1.86 5.39 4.43l.3 1.5 1.53.11c1.56.1 2.78 1.41 2.78 2.96 0 1.65-1.35 3-3 3zM8 13h2.55v3h2.9v-3H16l-4-4z"
                />
              </svg>
            </span>
            Top
          </button>
        </div>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-1goo8qo"
      >
        <div
          className="MuiBox-root css-5pmy0u"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
            data-testid="ForwardOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
            data-testid="ForwardOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
            />
          </svg>
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 css-c6mk4m-MuiTypography-root"
        >
          hmm... u/undefined
          hasn't posted recently
        </p>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-1bk1a37"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-reyiuj-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-e5mdo1-MuiSvgIcon-root"
          data-testid="ForwardOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiBox-root css-1kf8tao"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-f6ib2t-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="cover image"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          data-testid="cover-photo"
          height="94"
        />
        <div
          className="MuiBox-root css-fx6skf"
        >
          <div
            className="MuiBox-root css-k008qs"
          >
            <div
              className="MuiBox-root css-0"
            >
              <img
                alt="user photo"
                className="css-1kfbsc2"
              />
            </div>
          </div>
          <span
            className="MuiTypography-root MuiTypography-caption css-16o7n97-MuiTypography-root"
          >
            u/
          </span>
          <br />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-16dpvgk-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Create Your Own Avatar
            <span
              className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowForwardIosOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
            </span>
          </button>
          <div
            className="MuiBox-root css-17nleym"
          >
            <div
              className="MuiBox-root css-1fualzv"
            >
              <p
                className="MuiTypography-root MuiTypography-body2 css-7t6zra-MuiTypography-root"
              >
                Karma
              </p>
              <div
                className="MuiBox-root css-k008qs"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeString css-cf015h-MuiSvgIcon-root"
                  data-testid="FilterVintageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.7 12.4c-.28-.16-.57-.29-.86-.4.29-.11.58-.24.86-.4 1.92-1.11 2.99-3.12 3-5.19-1.79-1.03-4.07-1.11-6 0-.28.16-.54.35-.78.54.05-.31.08-.63.08-.95 0-2.22-1.21-4.15-3-5.19C10.21 1.85 9 3.78 9 6c0 .32.03.64.08.95-.24-.2-.5-.39-.78-.55-1.92-1.11-4.2-1.03-6 0 0 2.07 1.07 4.08 3 5.19.28.16.57.29.86.4-.29.11-.58.24-.86.4-1.92 1.11-2.99 3.12-3 5.19 1.79 1.03 4.07 1.11 6 0 .28-.16.54-.35.78-.54-.05.32-.08.64-.08.96 0 2.22 1.21 4.15 3 5.19 1.79-1.04 3-2.97 3-5.19 0-.32-.03-.64-.08-.95.24.2.5.38.78.54 1.92 1.11 4.2 1.03 6 0-.01-2.07-1.08-4.08-3-5.19zM12 16c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4z"
                  />
                </svg>
                <span
                  className="MuiTypography-root MuiTypography-caption css-15tzfi-MuiTypography-root"
                >
                  NaN
                </span>
              </div>
            </div>
            <div
              className="MuiBox-root css-1fualzv"
            >
              <p
                className="MuiTypography-root MuiTypography-body2 css-7t6zra-MuiTypography-root"
              >
                Cake Day
              </p>
              <div
                className="MuiBox-root css-k008qs"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeString css-cf015h-MuiSvgIcon-root"
                  data-testid="CakeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 6c1.11 0 2-.9 2-2 0-.38-.1-.73-.29-1.03L12 0l-1.71 2.97c-.19.3-.29.65-.29 1.03 0 1.1.9 2 2 2zm4.6 9.99-1.07-1.07-1.08 1.07c-1.3 1.3-3.58 1.31-4.89 0l-1.07-1.07-1.09 1.07C6.75 16.64 5.88 17 4.96 17c-.73 0-1.4-.23-1.96-.61V21c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-4.61c-.56.38-1.23.61-1.96.61-.92 0-1.79-.36-2.44-1.01zM18 9h-5V7h-2v2H6c-1.66 0-3 1.34-3 3v1.54c0 1.08.88 1.96 1.96 1.96.52 0 1.02-.2 1.38-.57l2.14-2.13 2.13 2.13c.74.74 2.03.74 2.77 0l2.14-2.13 2.13 2.13c.37.37.86.57 1.38.57 1.08 0 1.96-.88 1.96-1.96V12C21 10.34 19.66 9 18 9z"
                  />
                </svg>
                <span
                  className="MuiTypography-root MuiTypography-caption css-15tzfi-MuiTypography-root"
                >
                  Invalid date
                </span>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root css-1bnxa12"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-vskcn0-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Unfollow
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1xmowgn-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Chat
            </button>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-u3opfe-MuiButtonBase-root-MuiButton-root"
            data-testid="show-more"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            More options
          </button>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-f5ml2f-MuiPaper-root-MuiCard-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
        >
          Moderator of these communities
        </p>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ub553u-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        fontSize={15}
        fontWeight="bold"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Back to Top
      </button>
    </div>
  </div>
</div>
`;
